### ISSUE TITLE
fill_template.sh fails on macOS

### ISSUE THREAD

**Issue by msgilligan** [emilk/eframe_template#164]
The `fill_template.sh` does not work with the built-in `sed` command on macOS (`/usr/bin/sed`) and produces the following error: 

```
Patching files...
sed: 1: "Cargo.toml": invalid command code C
```

I worked around the issue by installing the Nix package `gnused`. Once that version of `sed` is first in my `$PATH` `fill_template.sh` works correctly.

**Comment by c-git** [comment]
I had redone the template in rust so no sed needed and had decided it maybe wasn't worth it as it wasn't that much better than the script. But I can revisit given considering cross platform, if that is actually helpful to someone other than me.

**Comment by msgilligan** [comment]
At the very least you could just document that macOS users need to install GNU sed. It was pretty easy to figure out, but a comment in the script or elsewhere would IMO resolve this issue and might be simpler than publishing a Rust version.

**Comment by c-git** [comment]
# TLDR

I suspect the original author may not have had experience with macOS and thus didn't know of the differences between the versions of `sed`. I think the simplest next step is to add a comment as you suggested.

# Problems with that solution

But there is also another version that had to be done for windows https://github.com/emilk/eframe_template/pull/153 and now technically we've doubled the number of versions to be maintained along with needing a comment about versions of `sed`. 

# Problems I can think of with a rust solution

Rust scripts haven't been stabilized yet and would require nightly to run them.  Otherwise I think they would be similar level of difficulty to doing a rust script compared with the current script.

The other option is doing a cargo subcommand but that would require the user to install a binary instead of just run a script again probably not worthwhile unless you start a lot of new projects.

# My previous attempt (that I think missed the mark in some regards)

The main reason I didn't make a PR for the version I did was because my information doesn't change much between projects and thus I did something a bit more complicated (which in hindsight was maybe a bit too complicated) that would reuse information. The current version of what I did is [online](https://github.com/c-git/my_egui_template/tree/main/setup_script) but I do not think it works well yet for the general case.

**Comment by saurabhkaul** [comment]
@msgilligan @c-git I ran into the same issue and thought this warranted a comment in the README.
Please let me know your thoughts here:  #187.

**Comment by c-git** [comment]
I think the update to the readme is the best short term mitigation. But I think once [rust's single file scripts](https://github.com/rust-lang/cargo/issues/12207) stabilize we should migrate over.
